src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -293,6 +293,83 @@ describe('WorkspacesUsecases', () => {
     });
   });
 
+  describe('setupWorkspace', () => {
+    const user = newUser();
+
+    it('When workspace does not exist, then it should throw', async () => {
+      jest.spyOn(workspaceRepository, 'findOne').mockResolvedValueOnce(null);
+
+      await expect(
+        service.setupWorkspace(user, 'workspace-id', {
+          name: 'Test Workspace',
+          encryptedMnemonic: 'encryptedMnemonic',
+        }),
+      ).rejects.toThrow(NotFoundException);
+    });
+
+    it('When workspace is setup, then it should add the owner as user to the workspace', async () => {
+      const workspace = newWorkspace();
+
+      jest
+        .spyOn(workspaceRepository, 'findOne')
+        .mockResolvedValueOnce(workspace);
+
+      await service.setupWorkspace(user, 'workspace-id', {
+        name: 'Test Workspace',
+        encryptedMnemonic: 'encryptedMnemonic',
+      });
+
+      expect(workspaceRepository.addUserToWorkspace).toHaveBeenCalled();
```

Comment:
Ensuring the function has been called with the owner's data would make this test stronger